py-polars/src/lazy/dataframe.rs

```diff
@@ -773,3 +807,34 @@ impl PyLazyFrame {
         self.ldf.clone().unnest(cols).into()
     }
 }
+
+struct JsonScan {
```

Comment:
I think we should add this to `polars-lazy` under the `json` feature flag.

Reply:
moved into polars-lazy & created a `LazyJsonReader`. I removed the `skip_rows`, as that isnt supported yet, but it should be trivial to implement at the reader level. 

I'll take another look at the csv impl to see how we could support predicate pushdowns too. right now only `n_rows` & projections are fully pushed down.

Reply:
> I'll take another look at the csv impl to see how we could support predicate pushdowns too. right now only n_rows & projections are fully pushed down.

If we can create an iterator over small batches we can use https://github.com/pola-rs/polars/blob/c1716348e85cfb9da83ffab38bec05c8e3b1b191/polars/polars-io/src/lib.rs#L114 to apply the optimizations.